frontend/src/utils/annotations.js

```diff
@@ -1,14 +1,37 @@
-export function generateBVOptions(options) {
+import {getValueFromKeyPath} from "@/utils/dict";
+
+export function generateBVOptions(options, document) {
     let optionsList = []
     if (Array.isArray(options)) {
-        for( let i in options){
-            const option = options[i]
-            optionsList.push({
-                value: option.value,
-                text: option.label
-            })
+        for (let option of options){
+            if (document && option && typeof option === "object"
+                && typeof option.fromDocument === "string") {
+                // one or more options taken from the document data
+                const propertyPath = option.fromDocument
+                // fromDocument is supposed to be a dot-separated path, but if
+                // the whole path is found as a top-level property name then use
+                // it, i.e. try opt['foo.bar.baz'] first, then opt.foo.bar.baz
+                // second
+                let propertyValue = (propertyPath in document) ?
+                    document[propertyPath] : getValueFromKeyPath(document, propertyPath);
+                optionsList.push(...generateBVOptions(propertyValue));
+            } else if (option !== null && typeof option !== "undefined") {
+                // single option
+                if (typeof option === 'string') {
```

Comment:
Technically this bit also allows the list-of-strings format within the _project_ configuration directly, i.e.

```json
{"type":"checkbox", "name":"foo", "options":["option 1", "option 2"]}
```

but this should be rejected by the schema-driven JSON editor.